src/main/java/aquality/selenium/core/waitings/IConditionalWait.java

```diff
@@ -0,0 +1,60 @@
+package aquality.selenium.core.waitings;
+
+import org.openqa.selenium.support.ui.ExpectedCondition;
+
+import java.util.Collection;
+import java.util.concurrent.TimeoutException;
+import java.util.function.BooleanSupplier;
+import java.util.function.Function;
+
+/**
+ * Utility used to wait for some condition.
+ */
+public interface IConditionalWait {
+
+    /**
+     * Wait for some condition within timeout. Method does not use WebDriverWait
+     * Default values for timeouts used from configuration settings file
+     *
+     * @param condition condition with boolean result (predicate)
+     * @param message   Part of error message in case of Timeout exception
+     * @return true if the condition has been met during the timeout
+     */
+    boolean waitForTrue(BooleanSupplier condition, String message);
```

Comment:
void type should be here and docs should tell about exception

Reply:
refactored